h/h_api/bulk_api/executor.py

```diff
@@ -0,0 +1,71 @@
+"""Implementations of an 'Executor' responsible for running bulk commands."""
+from h.h_api.bulk_api.model.report import Report
+from h.h_api.enums import CommandResult, CommandType
+
+
+class Executor:
+    """A callback to provide concrete functionality for actions in BulkJob."""
+
+    def configure(self, config):
+        """
+        Configure the job from the provided config object.
+
+        :param config: A config object
+        :return:
+        """
+        pass
+
+    def execute_batch(self, command_type, data_type, default_config, batch):
+        """
+        Execute the actions specified, returning any referenced ids.
+
+        The implementer is expected to:
+
+         * Full-fill the action
+         * Respect the configuration passed
+         * Return a list of `Report` objects in the same order as the items
+           in the batch
+
+        :param command_type: The CommandType being (e.g. UPSERT)
+        :param data_type: The DataType being modified
+        :param default_config: Configuration which applies to all actions
```

Comment:
```suggestion
        :param default_config: The Configuration model that applies to all actions
```

Reply:
This isn't quite right, it's a dict at this point. I'll change this to reflect that

Reply:
I've made this clearer now. This isn't the entire config object, it's only options deemed relevant to this action for the executor (plenty is handled for you).